replaying-share/src/test/java/com/jakewharton/rx/ReplayingShareObservableTest.java

```diff
@@ -190,4 +192,57 @@
     observableB.subscribe(observerB2);
     observerB2.assertValues("Bar");
   }
+
+  @Test public void completeClearsCacheAndResubscribes() {
+    List<String> start = new ArrayList<>();
+    start.add("initA");
+
+    PublishSubject<String> upstream = PublishSubject.create();
+    Observable<String> replayed = upstream.startWith(start).compose(ReplayingShare.<String>instance());
+
+    TestObserver<String> observer1 = new TestObserver<>();
+    replayed.subscribe(observer1);
+    observer1.assertValues("initA");
+
+    TestObserver<String> observer2 = new TestObserver<>();
+    replayed.subscribe(observer2);
+    observer1.assertValues("initA");
+
+    upstream.onComplete();
+    observer1.assertComplete();
+    observer2.assertComplete();
+
+    start.set(0, "initB");
+
+    TestObserver<String> observer3 = new TestObserver<>();
+    replayed.subscribe(observer3);
+    observer3.assertValues("initB");
+  }
+
+  @Test public void errorClearsCacheAndResubscribes() {
+    List<String> start = new ArrayList<>();
+    start.add("initA");
+
+    PublishSubject<String> upstream = PublishSubject.create();
+    Observable<String> replayed = upstream.startWith(start).compose(ReplayingShare.<String>instance());
+
+    TestObserver<String> observer1 = new TestObserver<>();
```

Comment:
You can do replayed.test() instead of manually creating the TestObserver and subscribing to it.

Reply:
Yes but then I would have to change all of the tests which I didn't want to bother with. Feel free to send a PR.